lib/spack/spack/test/cmd/env.py

```diff
@@ -34,8 +35,6 @@
 def check_mpileaks_install(viewdir):
     """Check that the expected install directories exist."""
     assert os.path.exists(str(viewdir.join('.spack', 'mpileaks')))
-    # Check that dependencies got in too
-    assert os.path.exists(str(viewdir.join('.spack', 'libdwarf')))
```

Comment:
@becker33: I reviewed again; everything looks good but this line.  Why was the check for `libdwarf` removed?

Note: This used to be spread over many tests, but @tldahlgren consolidated the logic, and I applied the change to the consolidated function here.  Are we checking that dependencies are linked in properly somewhere else?

Reply:
Because we changed the default semantics for environment views to only link the roots. See 5905b54. We discussed this about two weeks ago, based on a request from @lee218llnl

Reply:
@tgamblin 

Because we changed the default semantics for environment views to only link the roots. See 5905b54. We discussed this about two weeks ago, based on a request from @lee218llnl

Reply:
There are options to tailor this behavior in that commit, and tests for the behavior of both options.

Reply:
Default is still to link everything, right?

Reply:
Nevermind -- found it.  More change requests below.